### PR TITLE
Fixes Incorrect emit on `Array[number]?.()` #2347

### PR DESCRIPTION
**Pull request by alberrttt**

### CODE REVIEW

File: src/TSTransformer/nodes/transformOptionalChain.ts [hunk 1]
```diff
@@ -257,10 +259,13 @@ function transformOptionalChainInner(
 			if (item.callOptional) {
 				if (item.kind === OptionalChainItemKind.PropertyCall) {
 					baseExpression = luau.property(convertToIndexableExpression(baseExpression), item.name);
-				} else {
+				} else if (item.kind === OptionalChainItemKind.ElementCall) {
+					const [index] = state.capture(() => transformExpression(state, item.argumentExpression));
```
Comment (Fireboltofdeath) [inline src/TSTransformer/nodes/transformOptionalChain.ts:263]: Is there a reason you're capturing the prereqs here? And then discarding them?

Comment (alberrttt) [reply src/TSTransformer/nodes/transformOptionalChain.ts]: i forgot to remove that 
oops

File: src/TSTransformer/nodes/transformOptionalChain.ts [hunk 2]
```diff
@@ -257,10 +259,12 @@ function transformOptionalChainInner(
 			if (item.callOptional) {
 				if (item.kind === OptionalChainItemKind.PropertyCall) {
 					baseExpression = luau.property(convertToIndexableExpression(baseExpression), item.name);
-				} else {
+				} else if (item.kind === OptionalChainItemKind.ElementCall) {
+					const expType = state.typeChecker.getNonOptionalType(state.getType(item.expression.expression));
```
Comment (Dionysusnu) [inline src/TSTransformer/nodes/transformOptionalChain.ts:263]: This call is unnecessary - `addOneIfArrayType` already covers undefined as a possible type. Line 263old/267new seems like the only needed change.

File: src/TSTransformer/nodes/transformOptionalChain.ts [hunk 3]
```diff
@@ -257,10 +259,12 @@ function transformOptionalChainInner(
 			if (item.callOptional) {
 				if (item.kind === OptionalChainItemKind.PropertyCall) {
 					baseExpression = luau.property(convertToIndexableExpression(baseExpression), item.name);
-				} else {
+				} else if (item.kind === OptionalChainItemKind.ElementCall) {
```
Comment (Dionysusnu) [inline src/TSTransformer/nodes/transformOptionalChain.ts:262]: This kind is already the only remaining one in this statement - explicitly if-ing on it leaves an edge case if the kind would ever be expanded. I also don't see the need to change it for this PR specifically.

**Comment by alberrttt** [comment]
I messed up my git, I'm going to open another PR.

**Pull request closed without merging**